### PR TITLE
Protect: Improve manual scan flow

### PR DESCRIPTION
**Pull request by dkmyta**
#### Description
This PR addresses some inconsistencies in the UI flow when processing a manual scan.

#### Changes proposed in this Pull Request:

* Adds a `START_SCAN_OPTIMISTICALLY` action to manually update the status to `scanning`
* Updates the conditional check for the in-progress display to consider the current `status`
* Updates the UI flow when processing a successful manual scan: 
   * Scan now button click
   * Display success notice
   * Dismiss success notice after 1s delay
   * Manually assign `scanning` to status 
   * Immediately display `in-progress` screen
   * Begin refreshing status continuously after 5s delay
   * Render the display accordingly

#### Other information:
- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

- 1201069996155224-as-1203235765510738

#### Does this pull request change what data or activity we track or use?

- No

#### Testing instructions:

* Fire up a Jurassic Ninja instance w/ the Jetpack Beta plugin installed using this Protect branch
* Upgrade to Scan inclusive plan
* Queue a scan from the UI and verify that: 
   * The success notice only displays for ~1s
   * The in-progress screen displays immediately and for the duration that the scan is running
   * The new summary and threat list displays once the scan is complete and the Scan now button is no longer in a loading state
* Ensure all related actions remain functional and perform as expected

### CODE REVIEW

**Comment by nateweller** [comment]
Great PR! Code looks great, works great.

I think we could remove the entire "Scan was enqueued successfully" message and boot the user straight to the scanning screen, since we're forcing the 'scanning' status having it go to the in-progress screen will be confirmation that the button click worked, and then we don't have to make the user wait for a notice to appear and disappear. What do you think?

File: projects/plugins/protect/src/js/state/actions.js [hunk 1]
```diff
@@ -274,20 +260,23 @@ const scan = ( callback = () => {} ) => async ( { dispatch } ) => {
...
+					dispatch( clearNotice() );
 				}, 1000 );
 			} )
 			.then( () => {
-				return dispatch( refreshStatusUntilScanning() );
+				setTimeout( () => {
+					dispatch( startScanOptimistically() );
+				}, 2 * 1000 );
+				setTimeout( () => {
+					dispatch( refreshStatus( true ) );
```
Comment (nateweller) [inline projects/plugins/protect/src/js/state/actions.js]: Once we've manually set the scan status to 'scanning', we should in theory start polling `refreshStatus()` automatically in the `AdminPage`'s `useStatusPolling()` hook. If that is the case, we shouldn't need to dispatch the `refreshStatus()` action here.

**Review by nateweller — CHANGES_REQUESTED** [review]
Left some notes on:
- [x] Removing the notice / delay between button click and scan in progress screen.
- [x] Removing the `refreshStatus()` dispatch if the polling hook already handles this.

**Comment by dkmyta** [comment]
>I think we could remove the entire "Scan was enqueued successfully" message and boot the user straight to the scanning screen

Yep, good call - I think the in-progress screen is certainly confirmation enough that the manual scan has been initiated.

>Once we've manually set the scan status to 'scanning', we should in theory start polling refreshStatus() automatically

I'll test this out to confirm - I am not sure I ran through a scenario where there wasn't an additional refresh queued in some form after the success notice so if that is the case, that is fantastic!

**Review by nateweller — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**